src/frame/sdk/DocumentAdvancedApi.ts

```diff
@@ -0,0 +1,32 @@
+import * as DocumentOperations from "./DocumentOperations";
+import ApiState from "../ApiState";
+import Future from "futurejs";
+import {documentToByteParts} from "../FrameUtils";
+import SDKError from "../../lib/SDKError";
+import {ErrorCodes} from "../../Constants";
+import EncryptedDekEndpoints from "../endpoints/EncryptedDekEndpoints";
+
+interface DecryptedUnmanagedDocumentResponse {
+    data: Uint8Array;
+}
+/**
+ * Retrieves sharing/meta information for the provided document ID. Then decrypts the provided data using the current users key and returns formatted doc structure.
+ * @param {string}     documentID        Unique lookup key of document to decrypt
+ * @param {Uint8Array} encryptedDocument Data of document to decrypt
+ */
+export function decryptLocalDoc(edeks: Uint8Array, encryptedDocument: Uint8Array): Future<SDKError, DecryptedUnmanagedDocumentResponse> {
```

Comment:
Do we want to do any validation of the edek data? I don't know what the performance characteristics are of deserializing here, but performance concerns are the main downside that I can see.

Reply:
It looks like they're getting partially validated down below via the `validateEncryptedDeks` method that gets called before we get to the frame code here.

Reply:
If this is to be delayed until encrypt is implemented, it should be added as an AC on that ticket. From a dependency perspective, I think it makes more sense to add it now, but it's not a show-stopper for me. 

Again, I'm not sure it's worth doing at all due to performance, but I suspect the performance will be acceptable.